From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anonymous <anonymous@example.com>
Date: Thu, 15 Aug 2024 00:00:00 +0000
Subject: [PATCH] imx8mp: disable CAAM and enable legacy image format

Prevent the i.MX8M boards from initialising the CAAM hardware and allow
booting legacy U-Boot images.

* Stop Kconfig for i.MX8M SoCs from unconditionally selecting the CAAM
  hardware block. This lets individual boards opt out of CAAM support.
* Update the Debix defconfig to explicitly disable CAAM and enable
  legacy image handling in both U-Boot proper and the SPL.
* Enable legacy image support in the SPL for the EVK board so it can load
  legacy uImage-based flash binaries.

Signed-off-by: Anonymous <anonymous@example.com>
---
 arch/arm/mach-imx/imx8m/Kconfig        | 2 +-
 configs/imx8mp_debix_model_a_defconfig | 5 +++++
 configs/imx8mp_evk_defconfig           | 1 +
 3 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-imx/imx8m/Kconfig b/arch/arm/mach-imx/imx8m/Kconfig
index 2df470f..3a3c1d1 100644
--- a/arch/arm/mach-imx/imx8m/Kconfig
+++ b/arch/arm/mach-imx/imx8m/Kconfig
@@
-select HAS_CAAM
+imply HAS_CAAM
 select LTO
 select ROM_UNIFIED_SECTIONS
 select ARMV8_CRYPTO
diff --git a/configs/imx8mp_debix_model_a_defconfig b/configs/imx8mp_debix_model_a_defconfig
index 4954333..2adfb25 100644
--- a/configs/imx8mp_debix_model_a_defconfig
+++ b/configs/imx8mp_debix_model_a_defconfig
@@
 CONFIG_SPL=y
 CONFIG_SPL_IMX_ROMAPI_LOADADDR=0x48000000
 CONFIG_SYS_LOAD_ADDR=0x40480000
 CONFIG_FIT=y
 CONFIG_FIT_EXTERNAL_OFFSET=0x3000
 CONFIG_SPL_LOAD_FIT=y
+CONFIG_LEGACY_IMAGE_FORMAT=y
+CONFIG_SPL_LEGACY_IMAGE_SUPPORT=y
+# CONFIG_HAS_CAAM is not set
+# CONFIG_FSL_CAAM is not set
+# CONFIG_CMD_BLOB is not set
 CONFIG_SYS_BOOTM_LEN=0x2000000
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_SYSTEM_SETUP=y
diff --git a/configs/imx8mp_evk_defconfig b/configs/imx8mp_evk_defconfig
index 7112fdf..4954333 100644
--- a/configs/imx8mp_evk_defconfig
+++ b/configs/imx8mp_evk_defconfig
@@
 CONFIG_BOOTCOMMAND="run sr_ir_v2_cmd;run distro_bootcmd;run bsp_bootcmd"
 CONFIG_FIT=y
 CONFIG_FIT_EXTERNAL_OFFSET=0x3000
 CONFIG_SPL_LOAD_FIT=y
+CONFIG_SPL_LEGACY_IMAGE_SUPPORT=y
 CONFIG_REMAKE_ELF=y
 CONFIG_OF_BOARD_FIXUP=y
 CONFIG_OF_BOARD_SETUP=y
-- 
2.34.1
